lib/datadog/appsec/contrib/faraday/connection.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Datadog
+  module AppSec
+    module Contrib
+      module Faraday
+        # Handles installation of our middleware if the user has *not*
+        # already explicitly configured our middleware for this correction.
+        #
+        # Wraps Faraday::Connection#initialize:
+        # https://github.com/lostisland/faraday/blob/ff9dc1d1219a1bbdba95a9a4cf5d135b97247ee2/lib/faraday/connection.rb#L62-L92
+        module Connection
```

Comment:
I would like to ask for a better name. It's hard to understand what this is, should we use at least `ConnectionPatch`, but this name is also not perfect, but at least it answers the question "what is it"